ui/src/hooks/useZsentry.ts

```diff
@@ -210,19 +210,17 @@ export const useSentry = (lastSendTime?: Ref<Date>, t?: any): IUseSentry => {
                 .then(() => {
                     ZmodemBrowser.Browser.save_to_disk(buffer, xfer.get_details().name);
                     message.info(`${t('DownloadSuccess')}: ${detail.name}`);
-
                     terminal.write('\r\n');
-
-                    if (zmodeSession.value) {
-                        zmodeSession.value.abort();
-                    }
                 })
                 .catch((e: Error) => {
                     message.error(`Error: ${e}`);
                 });
         });
 
         zsession.on('session_end', () => {
+            if (zmodeSession.value) {
+                zmodeSession.value.abort();
+            }
             terminal.write('\r\n');
         });
 
```

Comment:
There does not seem to be any immediate issues with this JavaScript code. The code appears to handle a situation where two processes are communicating via `ZMODEN`, transferring data between them using `browser.saveToDisk()` function. This is useful for managing files that require transfer like backup copies of a user's documents or images.
However, there may be a few areas you can consider refactoring or improving if necessary:

1. The current implementation saves the file directly from ZMODEN without checking whether it's already exist inside the local disk. You should add some condition to verify that it was previously saved into the local system before saving again.

2. In asynchronous operations, ensure that error handling has been done properly and effectively. For instance, errors occurring in one operation might propagate through the process chain, so proper error propagation and recovery would improve robustness of the application.

3. Check the performance implications of using `browser.saveToDisk()`. If you're working with huge amounts of data, potentially slower write times due to browser-specific limitations could lead to long running times even during successful transfers.

The above points highlight certain best practices which will help avoid common pitfalls when implementing such complex communication systems.